curvefs/src/client/rpcclient/metaserver_client.cpp

```diff
@@ -516,12 +517,11 @@ MetaStatusCode MetaServerClientImpl::GetOrModifyS3ChunkInfo(
                          << ", errorcode: " << ret
                          << ", errmsg: " << MetaStatusCode_Name(ret);
             return ret;
-        } else if (response.has_appliedindex() &&
-            (!returnInode || response.has_inode())) {
+        } else if (response.has_appliedindex()) {
             metaCache_->UpdateApplyIndex(CopysetGroupID(poolID, copysetID),
                                          response.appliedindex());
-            if (returnInode) {
-                out->CopyFrom(response.inode());
+            if (returnS3ChunkInfoMap) {
+                out->swap(*response.mutable_s3chunkinfomap());
```

Comment:
you should check whether `out` is nullptr or not

Reply:
done